tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -534,4 +497,91 @@ void initTorchFunctions(PyObject* module) {
   }
 }
 
+PyObject* handle_torch_function(PythonArgs &r, PyObject* args, PyObject* kwargs, PyTypeObject &torch_api) {
+  PyObject* torch_api_function =
+    PyObject_FastGetAttrString((PyObject*)&torch_api, const_cast<char*>(r.get_func_name().data()));
+  TORCH_INTERNAL_ASSERT(torch_api_function != NULL, "torch API function must exist");
+  PyObject* ret = nullptr;
+  for (auto arg : r.overloaded_args) {
+    PyObject* torch_function = PyObject_FastGetAttrString(arg, "__torch_function__");
+    ret = PyObject_CallFunctionObjArgs(torch_function, torch_api_function, args, kwargs, NULL);
+    if (ret == Py_NotImplemented) {
```

Comment:
Conversion happens here.

Reply:
I originally had it this way, but then I needed to check if I was on the last iteration of the for loop or not and it seemed simpler to do it this way. In any case it looks like I'll be updating this to use pybind11 instead to handle the reference counting so this code will change.